.github/workflows/ci-go-cover.yml

```diff
@@ -14,7 +14,7 @@
 # 1. Change workflow name from "cover 100%" to "cover ≥92.5%". Script will automatically use 92.5%.
 # 2. Update README.md to use the new path to badge.svg because the path includes the workflow name.
 
-name: cover ≥85.0%
+name: cover ≥83.1%
```

Comment:
Any reason the cover is dropped ? If possible, can it be at least retained,

Reply:
There are now a couple more places where error checks are being performed during marshaling, but there are no straigh-forward way of triggering those paths (either due to prior checks, or because they come form the marshaling libs, that would basically never fail).

Reply:
Premise: I am not overly sensitive to dropping some coverage in a big refactoring PR like this (especially given there seems to be a good rationale for that), but ...

> [o]r because they come form the marshaling libs, that would basically never fail).

... could these cases be handled with `assert`s?

Reply:
Potentially, but that would be making assumptions about the bevior of said libs. It would also be more brittle w.r.t. any future refactors of this code.

Checking for error and returning it is the more robust (and "correct") behavior, even if it hard to intentionally trigger it.